projects/packages/password-checker/src/class-password-checker.php

```diff
@@ -11,128 +11,141 @@
-
-		/**
-		 * Filters Jetpack's password strength enforcement settings. You can supply your own passwords
-		 * that should not be used for authenticating in addition to weak and easy to guess strings for
-		 * each user. For example, you can add passwords from known password databases to avoid compromised
-		 * password usage.
-		 *
-		 * @since 7.2.0
-		 *
-		 * @param array $restricted_passwords strings that are forbidden for use as passwords.
-		 */
-		$this->common_passwords = apply_filters( 'jetpack_password_checker_restricted_strings', array() );
-
 		if ( is_null( $user ) ) {
 			$this->user_id = get_current_user_id();
 		} elseif ( is_object( $user ) && isset( $user->ID ) ) {
-
 			// Existing user, using their ID.
 			$this->user_id = $user->ID;
-
 		} elseif ( is_object( $user ) ) {
-
 			// Newly created user, using existing data.
 			$this->user    = $user;
 			$this->user_id = 'new_user';
-
 		} else {
 			$this->user_id = $user;
 		}
-		$this->min_password_length = apply_filters( 'better_password_min_length', $this->min_password_length );
+
+		/**
+		 * Filters the password strength enforcement settings.
+		 *
+		 * You can supply your own passwords that should not be used for authenticating in addition to weak and easy
+		 * to guess strings for each user. For example, you can add passwords from known password databases to avoid
+		 * compromised password usage.
+		 *
+		 * @param array $common_passwords strings that are forbidden for use as passwords.
+		 */
+		$this->common_passwords = apply_filters( 'password_checker_common_passwords', $this->common_passwords );
+
+		/**
+		 * Filters the password strength enforcement settings.
+		 *
+		 * You can modify the minimum password length using this filter.
+		 *
+		 * @param int $minimum_password_length minimum password length.
+		 */
+		$this->minimum_password_length = apply_filters( 'password_checker_minimum_password_length', $this->minimum_password_length );
+
+		/**
+		 * Filters the password strength enforcement settings.
+		 *
+		 * You can modify the minimum entropy bits requirement using this filter.
+		 *
+		 * @param int $minimum_entropy_bits minimum entropy bits requirement.
+		 */
+		$this->minimum_entropy_bits = apply_filters( 'password_checker_minimum_entropy_bits', $this->minimum_entropy_bits );
 	}
 
 	/**
 	 * Run tests against a password.
 	 *
-	 * @param String  $password      the tested string.
-	 * @param Boolean $required_only only test against required conditions, defaults to false.
-	 * @return array $results an array containing failed and passed test results.
+	 * @param string $password      the tested string.
+	 * @param bool   $required_only only test against required conditions, defaults to false.
+	 *
+	 * @return array an array containing failed and passed test results.
 	 */
 	public function test( $password, $required_only = false ) {
-
+		// Save the password for later use.
 		$this->password = $password;
```

Comment:
Hmm. For safety you should probably clear this field before the method returns, just to reduce the possibility that someone might be able to dump this object and recover the password later. Yeah, that's a bit paranoid, but paranoia is good when handling passwords.

```php
$this->password = $password;
try {
    ... the whole rest of the method ...
} finally {
    $this->password = null;
}
```

Or just pass the password through to the tests as an argument.

Reply:
I went with passing the $password as an argument.
That's a great point about dumping the object. I think it would also be true if the object were serialized for some reason.